In [1]:
import numpy as np
import pandas as pd

np.random.seed(406)

n = 5000
df = pd.DataFrame({
    'favorite_animal': np.random.choice(['cat', 'dog', 'frog', 'lemur', 'panda'], n),
    'favorite_vegetable': np.random.choice(['brussel sprouts', 'potato', 'squash'], n),
    'favorite_fruit': np.random.choice(['banana', 'apple', 'blueberries'], n),
    'wears_glasses': np.random.choice(['yes', 'no'], n),
    'netflix_consumption': np.random.normal(10, 2, n),
    'open_browser_tabs': np.random.randint(2, 90, n),
})

In [2]:
df

favorite_animal favorite_vegetable favorite_fruit wears_glasses  \
0              lemur             potato          apple           yes   
1              panda             potato          apple           yes   
2                cat             squash    blueberries           yes   
3              lemur             squash          apple            no   
4                dog    brussel sprouts          apple           yes   
...              ...                ...            ...           ...   
4995            frog             potato    blueberries           yes   
4996            frog             potato          apple           yes   
4997           lemur             squash         banana            no   
4998           lemur             squash    blueberries           yes   
4999             dog             potato         banana           yes   

      netflix_consumption  open_browser_tabs  
0                8.313351                 44  
1               11.801073                 10  
2               10.105141                 35  
3               11.024605                 70  
4                6.732698                 73  
...                   ...                ...  
4995             9.992337                 59  
4996             8.916902                 46  
4997             7.317019                 70  
4998            12.720892                 50  
4999            12.890119                 53  

[5000 rows x 6 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   favorite_animal      5000 non-null   object 
 1   favorite_vegetable   5000 non-null   object 
 2   favorite_fruit       5000 non-null   object 
 3   wears_glasses        5000 non-null   object 
 4   netflix_consumption  5000 non-null   float64
 5   open_browser_tabs    5000 non-null   int64  
dtypes: float64(1), int64(1), object(4)
memory usage: 234.5+ KB


In [4]:
df.describe()

netflix_consumption  open_browser_tabs
count          5000.000000        5000.000000
mean              9.962152          45.949800
std               2.018416          25.380199
min               2.130308           2.000000
25%               8.570916          25.000000
50%               9.950686          46.000000
75%              11.348473          68.000000
max              17.534819          89.000000

# Questions:

### - What is the highest amount of netflix consumption? 17.535

In [8]:
most_netflix = df.netflix_consumption.max()
print(f'The highest amount of netflix consumptions\n is: {round(most_netflix,3)}')

The highest amount of netflix consumptions
 is: 17.535


In [14]:
from env import Percent

### - How many people wear glasses? What percentage of people is this? 2555, .511

In [10]:
df.wears_glasses.value_counts(dropna=False)

yes    2555
no     2445
Name: wears_glasses, dtype: int64

In [12]:
len(df[df.wears_glasses == 'yes'])

2555

In [15]:
Percent(len(df[df.wears_glasses == 'yes'])/len(df))

51.10%

### - How many people's favorite animal is a dog? 1002

In [17]:
df.favorite_animal.value_counts(dropna=False)

lemur    1028
frog     1015
dog      1002
cat       980
panda     975
Name: favorite_animal, dtype: int64

In [18]:
len(df[df.favorite_animal == 'dog'])

1002

### - What is the most common favorite animal? lemur

In [22]:
df.favorite_animal.mode()[0].upper()

'LEMUR'

### - What is the average netflix consumption for people that prefer brussel sprouts? 10.008

In [23]:
df.favorite_vegetable.value_counts(dropna=False)

brussel sprouts    1696
squash             1662
potato             1642
Name: favorite_vegetable, dtype: int64

In [26]:
round(df.netflix_consumption[df.favorite_vegetable == 'brussel sprouts'].mean(),3)

10.008

### - What is the most common favorite fruit for people who wear glasses and have more than 40 open browser tabs? blueberries

In [31]:
df.favorite_fruit[(df.open_browser_tabs > 40) & (df.wears_glasses == 'yes') ].value_counts()

blueberries    498
apple          477
banana         472
Name: favorite_fruit, dtype: int64

In [32]:
df.favorite_fruit[(df.open_browser_tabs > 40) & (df.wears_glasses == 'yes') ].max()

'blueberries'